custom_components/myskoda/sensor.py

```diff
@@ -295,6 +295,11 @@ def icon(self) -> str:  # noqa: D102
                 return "mdi:power-plug-battery"
         return "mdi:power-plug"
 
+    @property
+    def available(self):  # noqa: D102
+        if status := self._status():
+            return status.state is not None
```

Comment:
This will never evaluate to `False` because the `if` condition won't be entered if `self._status()` effectively returns `None`. In that case the function will end up returning `None` (the default) instead of a bool.

I think we just need `return self._status() is not None`

Reply:
Ah I read over the `status.state`

Reply:
I  guess just add a default `return False` at the end and we're good

```
    def available(self) -> bool:  # noqa: D102
        if status := self._status():
            return status.state is not None
        return False
```

Do we even need this if we fix `native_value()` so it doesn't return a stringified lowercase `None`? :)

```
    def native_value(self) -> str | None:  # noqa: D102
        if status := self._status():
            if status.state:
                return str(status.state).lower()
```

Reply:
Yep, thats an equally good solution

Reply:
I will test with this line of code tonight

Reply:
This works in my setup:

````    
def native_value(self) -> str | None:  # noqa: D102
        if status := self._status():
            if status.state:
                return str(status.state).lower()